bundles/net.tourbook/src/net/tourbook/tour/ActionDeleteMarkerDialog.java

```diff
@@ -0,0 +1,132 @@
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.tour;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import net.tourbook.Messages;
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.data.TourData;
+import net.tourbook.data.TourMarker;
+import net.tourbook.ui.ITourProvider;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.widgets.Display;
+
+public class ActionDeleteMarkerDialog extends Action {
+
+   private ITourProvider         _tourProvider;
+   private ArrayList<TourMarker> _tourMarkers;
+
+   /**
+    * @param tourProvider
+    */
+   public ActionDeleteMarkerDialog(final ITourProvider tourProvider) {
+
+      _tourProvider = tourProvider;
+
+      setText(Messages.app_action_delete_tour_marker);
+      setImageDescriptor(TourbookPlugin.getImageDescriptor(Messages.Image__delete_tour_marker));
+
+      setEnabled(false);
+   }
+
+   private static void doAction(final ITourProvider tourProvider,
+                                final ArrayList<TourMarker> selectedTourMarkers) {
+
+      final ArrayList<TourData> selectedTours = tourProvider.getSelectedTours();
+
+      // check if one tour is selected
+      if (selectedTours == null || selectedTours.size() != 1 || selectedTours.get(0) == null ||
+            selectedTourMarkers == null || selectedTourMarkers.size() == 0 || selectedTourMarkers.get(0) == null) {
+         return;
+      }
+
+      final TourData tourData = selectedTours.get(0);
+
+      if (tourData.isManualTour()) {
+         // a manually created tour do not have time slices -> no markers
+         return;
+      }
+
+      String dialogTitle = Messages.Dlg_TourMarker_MsgBox_delete_marker_title;
+      String dialogMessage = NLS.bind(Messages.Dlg_TourMarker_MsgBox_delete_marker_message, (selectedTourMarkers.get(0)).getLabel());
+
+      if (selectedTourMarkers.size() > 1) {
+         dialogTitle = Messages.Dlg_TourMarker_MsgBox_delete_markers_title;
+
+         final StringBuilder markersNames = new StringBuilder();
+         for (final TourMarker tourMarker : selectedTourMarkers) {
+            if (markersNames.toString().isEmpty() == false) {
+               markersNames.append(", "); //$NON-NLS-1$

```

Comment:
Newlines may be better readable


![2019-08-27_145627](https://user-images.githubusercontent.com/1283445/63773582-bd401f80-c8db-11e9-86c5-0d59b636bf07.png)

![2019-08-27_145913](https://user-images.githubusercontent.com/1283445/63773600-c630f100-c8db-11e9-912a-1e611ad4cffd.png)